commands/channel.go

```diff
@@ -345,32 +350,43 @@ func makeChannelPrivateCmdF(c client.Client, cmd *cobra.Command, args []string)
 }
 
 func renameChannelCmdF(c client.Client, cmd *cobra.Command, args []string) error {
-	var newDisplayName, newChannelName string
+	existingTeamChannel := args[0]
 
-	if len(args) < 2 {
-		return errors.New("Not enough arguments.")
+	newChannelName, err := cmd.Flags().GetString("name")
+	if err != nil {
+		return err
 	}
 
-	channel := getChannelFromChannelArg(c, args[0])
-	if channel == nil {
-		return errors.New("Unable to find channel '" + args[0] + "'")
+	newDisplayName, err := cmd.Flags().GetString("display_name")
+	if err != nil {
+		return err
 	}
 
-	newChannelName = args[1]
-	newDisplayName, errdn := cmd.Flags().GetString("display_name")
-	if errdn != nil {
-		return errdn
+	// Atleast one of display name or name flag must be present
+	if newDisplayName == "" && newChannelName == "" {
+		return errors.New("Require atleast one flag to rename channel, either 'name' or 'display_name'")
 	}
 
-	channelPatch := model.ChannelPatch{Name: &newChannelName}
+	channel := getChannelFromChannelArg(c, existingTeamChannel)
+	if channel == nil {
+		return errors.New("Unable to find channel from '" + existingTeamChannel + "'")
+	}
+
+	channelPatch := &model.ChannelPatch{}
+	if newChannelName != "" {
+		channelPatch.Name = &newChannelName
+	}
 	if newDisplayName != "" {
 		channelPatch.DisplayName = &newDisplayName
 	}
 
-	if _, response := c.PatchChannel(channel.Id, &channelPatch); response.Error != nil {
-		return response.Error
+	// Using PatchChannel API to rename channel
+	_, response := c.PatchChannel(channel.Id, channelPatch)
+	if response.Error != nil {
+		return errors.New("Cannot rename channel '" + channel.Name + "', error : " + response.Error.Error())
 	}
 
+	printer.Print("'" + channel.Name + "' channel renamed")
```

Comment:
As this is an update action, we should print the updated channel in case the user requests the JSON format. So if we modify the patch line to get the updated channel:

```go
updatedChannel, response := c.PatchChannel(channel.Id, channelPatch)
```

We can modify the print to be:

```suggestion
	printer.PrintT("'{{.Name}}' channel renamed", updatedChannel)
```